app/blueprints/instagram_post_blueprint.rb

```diff
@@ -14,7 +14,7 @@ class InstagramPostBlueprint < Blueprinter::Base
 
   field :image_files do |post|
     to_return = nil
-    unless post.image_file_names.nil?
+    unless post.image_file_names.nil? && post.aws_image_keys.blank?
```

Comment:
Sanity check: this is correct? It looks like everywhere else we're inverting this (from `unless …nil?` to `if …nil? == false`) when adding the blank-key check.